File: Project5_5/app/src/main/java/com/example/project5_5/MainActivity.java
package com.example.project5_5;


import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText edit1, edit2;
    Button btnAdd, btnSub,btnMul, btnDiv, btnRem;
    TextView textResult;
    String num1, num2;
    Integer result;
    Button[] numButtons=new Button[10];
    Integer[] numBtnIDs={R.id.BtnNum0, R.id.BtnNum1, R.id.BtnNum2,
            R.id.BtnNum3, R.id.BtnNum4, R.id.BtnNum5, R.id.BtnNum6,
            R.id.BtnNum7, R.id.BtnNum8, R.id.BtnNum9};
    int i;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle("그리드레이아웃 계산기");

        edit1 = (EditText) findViewById(R.id.Edit1);
        edit2 = (EditText) findViewById(R.id.Edit2);
        btnAdd = (Button) findViewById(R.id.BtnAdd);
        btnSub = (Button) findViewById(R.id.BtnSub);
        btnMul = (Button) findViewById(R.id.BtnMul);
        btnDiv = (Button) findViewById(R.id.BtnDiv);
        btnRem = (Button) findViewById(R.id.BtnRem);
        textResult = (TextView) findViewById(R.id.TextResult);


        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                num1 = edit1.getText().toString();
                num2 = edit2.getText().toString();

                if (num1.isEmpty() || num2.isEmpty()){
                    Toast.makeText(getApplicationContext(),
                            "빈칸 채워 주세요", Toast.LENGTH_SHORT).show();

                } else {
                    result = Integer.parseInt(num1) + Integer.parseInt(num2);
                    textResult.setText("더하기 계산 결과 : " + result.toString());
                }
            }

        });

        btnSub.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                num1 = edit1.getText().toString();
                num2 = edit2.getText().toString();

                if (num1.isEmpty() || num2.isEmpty()){
                    Toast.makeText(getApplicationContext(),
                            "빈칸 채워 주세요", Toast.LENGTH_SHORT).show();

                } else {
                    result = Integer.parseInt(num1) - Integer.parseInt(num2);
                    textResult.setText("빼기 계산 결과 : " + result.toString());
                }
            }

        });

        btnMul.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                num1 = edit1.getText().toString();
                num2 = edit2.getText().toString();

                if (num1.isEmpty() || num2.isEmpty()) {
                    Toast.makeText(getApplicationContext(),
                            "빈칸 채워 주세요", Toast.LENGTH_SHORT).show();
                } else {
                    result = Integer.parseInt(num1) * Integer.parseInt(num2);
                    textResult.setText("곱하기 계산 결과 : " + result.toString());
                }
            }


        });

        btnDiv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                num1 = edit1.getText().toString();
                num2 = edit2.getText().toString();

                if (num1.isEmpty() || num2.isEmpty()){
                    Toast.makeText(getApplicationContext(),
                            "빈칸 채워 주세요", Toast.LENGTH_SHORT).show();

                } else {
                    result = Integer.parseInt(num1) / Integer.parseInt(num2);
                    textResult.setText("나누기(몫) 계산 결과 : " + result.toString());
                }
            }

        });

        btnRem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                num1 = edit1.getText().toString();
                num2 = edit2.getText().toString();

                if (num1.isEmpty() || num2.isEmpty()){
                    Toast.makeText(getApplicationContext(),
                            "빈칸 채워 주세요", Toast.LENGTH_SHORT).show();

                } else {
                    result = Integer.parseInt(num1) % Integer.parseInt(num2);
                    textResult.setText("나머지 계산 결과 : " + result.toString());
                }
            }

        });


        for (i = 0; i < numBtnIDs.length; i++)
        {
            numButtons[i] = (Button) findViewById(numBtnIDs[i]);
        }
        for (i = 0; i < numBtnIDs.length; i++) {
            final int index;
            index = i;

            numButtons[index].setOnClickListener(new View.OnClickListener() {
                public void onClick(View v) {
                    if (edit1.isFocused()) {
                        num1 = edit1.getText().toString()
                                + numButtons[index].getText().toString();
                        edit1.setText(num1);
                    } else if (edit2.isFocused()) {
                        num2 = edit2.getText().toString()
                                + numButtons[index].getText().toString();
                        edit2.setText(num2);
                    } else {
                        Toast.makeText(getApplicationContext(), "먼저 에디트텍스트를 입력하세요.", Toast.LENGTH_SHORT).show();
                    }
                }
            });
        }
    }
}

File: PigChicken/app/src/main/java/com/example/androidsample/MainActivity.kt
package com.example.androidsample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import kotlinx.android.synthetic.main.activity_main.*
import android.content.Intent
import java.lang.Exception
import java.lang.NumberFormatException


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun converCurrency(view: View) {
        try {
            if (value1.text.isNotEmpty() && value2.text.isNotEmpty()) {

                val pig = Pig(value1)
                val chicken = Chicken(value2)


                if (pig == 0) {
                    textView.text = getString(R.string.re_enter4)
                } else textView.text = pig.toString() + " 마리"


                if (chicken == 0) {
                    textView2.text = getString(R.string.re_enter2)
                } else textView2.text = chicken.toString() + " 마리"

                textView3.text = ""
            } else textView3.text = getString(R.string.no_value_String)
        } catch (e: NumberFormatException) {
            textView3.text = "정수값 최대치는 2147483647 입니다"
        }

    }

    fun Pig(a: EditText): Int {
            val calcuvalue = a.text.toString().toInt()

            if ((calcuvalue % 4) != 0) {
                return 0
            } else if (calcuvalue is Int) {
                val Pigreg = calcuvalue / 4
                return Pigreg
            }

        return 0
    }

    fun Chicken(a: EditText): Int {
            val calcuvalue = a.text.toString().toInt()

            if ((calcuvalue % 2) != 0) {
                return 0
            } else if (calcuvalue is Int) {
                val Chickenwing = calcuvalue / 2
                return Chickenwing
            }

        return 0
    }
}